compiler/src/dotty/tools/dotc/transform/PostTyper.scala

```diff
@@ -75,6 +75,13 @@ class PostTyper extends MacroTransform with InfoTransformer { thisPhase =>
 
   override def changesMembers: Boolean = true // the phase adds super accessors and synthetic members
 
+  /**
+   * Serializable and AbstractFunction are added for scala2-library companion object of case class
+   *
+   * Ideally `compilingScala2StdLib` should be used, but it is initialized too late to be effective.
```

Comment:
We could try to move these three statements in Run.compileUnits earlier in front of `    ctx.base.usePhases(phases)`:

    val runCtx = ctx.fresh
    runCtx.setProfiler(Profiler())
    unfusedPhases.foreach(_.initContext(runCtx))

That should allow us to use `compilingScala2StdLib`.